src/ui/FileIcon/FileIcon.spec.tsx

```diff
@@ -0,0 +1,24 @@
+import { render, screen } from '../../testing';
+
+import { FileIcon } from './FileIcon';
+import { FileIconBadge } from './FileIconBadge';
+
+const CHILDREN_TEXT = 'DOCS';
+const BG_COLOR = 'red';
+
+describe('FileIcon', () => {
+  it('Should display children text', async () => {
+    render(<FileIcon>{CHILDREN_TEXT}</FileIcon>);
+    expect(screen.queryByText(CHILDREN_TEXT)).toBeInTheDocument();
+  });
+
+  it('Should display badge component', async () => {
+    render(
+      <FileIcon>
+        <FileIconBadge color="red">{CHILDREN_TEXT}</FileIconBadge>
+      </FileIcon>
+    );
+    expect(screen.queryByText(CHILDREN_TEXT)).toBeInTheDocument();
+    expect(screen.getByTestId('color')).toHaveStyle(`background-color: ${BG_COLOR}`);
```

Comment:
Тут такой момент, вот что в доках Testing Library говорится про `getByTestId`: 

In the spirit of the guiding principles, it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names.

Т.е. если есть возможность, то нужно использовать `getByText` и схожие, т.к. это то что в конце концов видит пользователь.

```suggestion
    expect(screen.getByText(CHILDREN_TEXT)).toHaveStyle(`background-color: ${BG_COLOR}`);
```